sds_data_manager/lambda_code/SDSCode/indexer.py

```diff
@@ -215,16 +245,94 @@ def batch_event_handler(event):
     dict
         HTTP response
     """
-
-    if event["detail-type"] != "Batch Job State Change":
-        return http_response(status_code=400, body="Invalid event")
+    # splited command looks like this:
+    # [
+    #   '',
+    #   'instrument codice ',
+    #   'level l1a ',
+    #   "s3_uri 's3://data-bucket/path/filename'",
+    #   "dependency [{'instrument': 'hit', 'level': 'l0', 'version': 'v00-01'}]"
+    #  ]
+    command = event["detail"]["container"]["command"][0].split("--")
+
+    # Get event inputs ready
+    s3_uri = command[3].replace("s3_uri '", "").replace("'", "")
+    filename = os.path.basename(s3_uri)
+
+    # TODO: post demo, revisit this and improve it
     if event["detail"]["status"] == "SUCCEEDED":
-        # TODO:
-        # Get filename and then
-        # write it to file catalog table
-        pass
+        # Frist write to status table and then
+        # write to file catalog with foreign key
+        # information
+        try:
+            # query and update status table record with new
+            # information from batch
+            with Session(db.get_engine()) as session:
+                # Had to query this way because the select statement
+                # returns a RowProxy object when it executes it,
+                # not the actual StatusTracking model instance,
+                # which is why it can't update table row directly.
+                result = (
+                    session.query(models.StatusTracking)
+                    .filter(models.StatusTracking.file_to_create_path == s3_uri)
+                    .first()
+                )
+
+                # update three fields with updated information
+                result.status = models.Status.SUCCEEDED
+                result.job_definition = event["detail"]["jobDefinition"]
+                result.ingestion_date = get_file_creation_date(s3_uri)
+                session.commit()
+
+                # Then write to file catalog table
+                sci_file = ScienceFilepathManager(filename)
+                metadata_params = sci_file.get_file_metadata_params()
+                metadata_params["file_path"] = s3_uri
+                metadata_params["status_tracking_id"] = result.id
+                update_file_catalog_table(metadata_params)
+                # TODO: send event to batch starter with filename information
+                # in upcoming PR. This is last component needed for demo.
+                # Event input will looks something like this:
+                # event = {
+                #     "Source": "imap.lambda",
+                #     "DetailType": "Processed File",
+                #     "Detail": {
+                #         "object": {
+                #               "key": filename
+                #         },
+                #     },
+                # }
+        except Exception as e:
+            logger.info(str(e))
+            return http_response(status_code=400, body=str(e))
```

Comment:
minor nit, but if you are catching an exception it should probably be logged at the error level to really catch our eyes.